### PR TITLE
Storybook: Add documentation to PluginSignatureBadge story

### LINKED ISSUE #51566: Chore: Improve `PluginSignatureBadge` story
- [ ] Add documentation in the form of `PluginSignatureBadge.mdx`
  - Where it's used in the product
  - What it's used for
  - What the various statuses represent

### PR DESCRIPTION
**Pull request by eledobleefe**
**What this PR does / why we need it**: It adds information about the Plugin Signature Badge component such as where the component is used or what its various statuses represent.

**Which issue(s) this PR fixes**: 

Fixes #51566 

**Special notes for your reviewer**: Is any information missing or incorrect?

### CODE REVIEW

File: packages/grafana-ui/src/components/PluginSignatureBadge/PluginSignatureBadge.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Meta, Preview, Props } from '@storybook/addon-docs/blocks';
+import { PluginSignatureBadge } from './PluginSignatureBadge';
+
+<Meta title="MDX|PluginSignatureBadge" component={PluginSignatureBadge} />
+
+# Plugin Signature Badge
+
+The Plugin Signature Badge can be found in each information plugin card of the configuration plugins pannel (Grafana menu > Configuration > Plugins).
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/PluginSignatureBadge/PluginSignatureBadge.mdx:8]:
```suggestion
The Plugin Signature Badge can be found in each information plugin card of the configuration plugins panel (Grafana menu > Configuration > Plugins).
```

File: packages/grafana-ui/src/components/PluginSignatureBadge/PluginSignatureBadge.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+This is a security measure to make sure plugins haven't been tampered with. Upon loading, Grafana checks to see the plugin signature status.
+
+## Badges:
+
+1.- Core: this badge indicates that it has been built into Grafana.
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/PluginSignatureBadge/PluginSignatureBadge.mdx:13]:
```suggestion
1.- Core: this badge indicates that the plugin has been built into Grafana.
```

File: packages/grafana-ui/src/components/PluginSignatureBadge/PluginSignatureBadge.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+<Preview>
+  <PluginSignatureBadge status="internal" />
+</Preview>
+
+2.- Signed: its signature was successfully verified.
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/PluginSignatureBadge/PluginSignatureBadge.mdx:19]:
```suggestion
2.- Signed: the plugin's signature was successfully verified.
```

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Nice job 🎉 Just a few suggestions to make the text more clear. But more importantly, does this now work on your machine? Because it does in mine!

**Comment by eledobleefe** [comment]
@JoaoSilvaGrafana thank you for your suggestions. And yes! It works now :)

**Review by JoaoSilvaGrafana — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
looks great @eledobleefe 🥳 

i fixed betterer due to the PR being created before the new betterer checks were introduced 👍

**Pull request merged**